### ISSUE TITLE
How to search the all sentence and not each word of sentence?

### ISSUE THREAD

**Issue by Ziadjb** [spatie/laravel-searchable#61]

**Comment by Design-Depot** [comment]
I think i was struggling with the same problem as you. I wanted to search in a column that has spaces ( unfortunately ). 

The problem was: if i searched a combination of words like: "new company:, the package would search with an 'or' query. Like:

`select
  *
from
  'my_table'
where
  (
    LOWER('title') LIKE "%new%"
    or LOWER('title') LIKE "%company%"
  )
  and 'my_table'.'deleted_at;' is null
order by
  'title' asc
`

However, i needed to search the full concatenated title so i wouldn't get a lot of results back that had similar names in the title. I managed to do a simple `str_replace` to replace the spaces in my search with percentage signs to use as a wildcard:

`
$results = (new Search())->registerModel(Model::class, function(ModelSearchAspect $modelSearchAspect) {
			$modelSearchAspect->addSearchableAttribute('title')
			                  ->orderBy('title');
		})->search(str_replace(' ', '%', $request->input('query')));
return response()->json($results);
`

This returns the search a lot better:

`
select
  *
from
 'my_table'
where
  (LOWER('title') LIKE "%new%company%")
  and 'my_table'.'deleted_at' is null
order by
  'title' asc
`

Long story short: don't use spaces in your SQL columns, but if you have to use them, this might solve the search issues.

**Comment by marcelloinfoweb** [comment]
Unfortunately it didn't work, by name it doesn't return anything, just by email. It could be with quotes, if you want with compound words.

**Comment by arctic-ice-cool** [comment]
This has been broken by https://github.com/spatie/laravel-searchable/commit/b9677b6a6302f5d154098b0abff77bd4bb23ac4b#diff-ae380cb7140a107b875f6ae035fe1b713b4d9e865cc244dcad44f13f325cfab1

Swapping:

```
    protected function addSearchConditions(Builder $query, string $term)
    {
        $attributes = $this->attributes;
        $searchTerms = explode(' ', $term);

        $query->where(function (Builder $query) use ($attributes, $term, $searchTerms) {
            foreach (Arr::wrap($attributes) as $attribute) {
                $sql = "LOWER({$query->getGrammar()->wrap($attribute->getAttribute())}) LIKE ? ESCAPE ?";

                foreach ($searchTerms as $searchTerm) {
                    $searchTerm = mb_strtolower($searchTerm, 'UTF8');
                    $searchTerm = str_replace("\\", $this->getBackslashByPdo(), $searchTerm);
                    $searchTerm = addcslashes($searchTerm, "%_");

                    $attribute->isPartial()
                        ? $query->orWhereRaw($sql, ["%{$searchTerm}%", '\\'])
                        : $query->orWhere($attribute->getAttribute(), $searchTerm);
                }
            }
        });
    }
```

back for:

```
    protected function addSearchConditions(Builder $query, string $term)
    {
        $attributes = $this->attributes;
        $searchTerms = explode(' ', $term);

        $query->where(function (Builder $query) use ($attributes, $term, $searchTerms) {
            foreach (Arr::wrap($attributes) as $attribute) {
                foreach ($searchTerms as $searchTerm) {
                    $sql = "LOWER({$attribute->getAttribute()}) LIKE ?";
                    $searchTerm = mb_strtolower($searchTerm, 'UTF8');

                    $attribute->isPartial()
                        ? $query->orWhereRaw($sql, ["%{$searchTerm}%"])
                        : $query->orWhere($attribute->getAttribute(), $searchTerm);
                }
            }
        });
    }
```

Seems to fix the issue.

With this commit, it appears that searching for strings with spaces returns no results?
